### PR TITLE
Add Microsoft.SignalRService/signalr 2022-02-01

### PR DESCRIPTION
**Pull request by ArchangelSDY**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
      - [x] Before 2022-02-28
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
      - [x] Before 2022-02-28
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by pilor** [comment]
>         "certificateName": {

if this is referencing a customCertificates resource it should use the full resource ID

---
Refers to: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/signalr.json:1690 in 02c719c. [](commit_id = 02c719c50203527860af8635fb6e2fa23144cc59, deletion_comment = False)

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/examples/SignalRCustomDomains_CreateOrUpdate.json
Comment (pilor) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/examples/SignalRCustomDomains_CreateOrUpdate.json:22]: if you are using the resource name as the indication of the domain name you will only ever be able to support domain names that are valid resource names (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#arguments-for-crud-on-resource).  FYI as this may impact you down the road

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/signalr.json
Comment (wanlwanl) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/signalr.json]: Hi @pilor , thanks for your comments! We will follow the string version in the future. There's something related I need to mention, please note that we already have a property [ResourceLogConfiguration.ResourceLogCategory](https://github.com/Azure/azure-rest-api-specs/blob/02c719c50203527860af8635fb6e2fa23144cc59/specification/signalr/resource-manager/Microsoft.SignalRService/stable/2021-10-01/signalr.json#L1652), the same format to [LiveTraceConfiguration.LiveTraceCategory](https://github.com/Azure/azure-rest-api-specs/blob/02c719c50203527860af8635fb6e2fa23144cc59/specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/signalr.json#L1745). The `category` shares the same concept and format. If customers use feature `resource log` ( and configure `ResourceLogCategory`), they will have chance to use another feature `live trace` ( and configure `LiveTraceCategory`). I have a concern that If the `enabled` changed to `status`, they will feel confused why these two are in different formats, maybe keep the true/false is not bad in this scenario. What do you think?

**Comment by ArchangelSDY** [comment]
> > ```
> >     "certificateName": {
> > ```
> 
> if this is referencing a customCertificates resource it should use the full resource ID
> 
> Refers to: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/signalr.json:1690 in [02c719c](https://github.com/Azure/azure-rest-api-specs/commit/02c719c50203527860af8635fb6e2fa23144cc59). [](commit_id = [02c719c](https://github.com/Azure/azure-rest-api-specs/commit/02c719c50203527860af8635fb6e2fa23144cc59), deletion_comment = False)

Updated with:

```
"customCertificate": {
    "id": "<custom-certificate-resource-id>"
}
```

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/examples/SignalRCustomDomains_CreateOrUpdate.json
Comment (ArchangelSDY) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/examples/SignalRCustomDomains_CreateOrUpdate.json]: Nice catch. Added a dedicated property `domainName`.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/signalr.json
Comment (pilor) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2022-02-01/signalr.json:1758]: if its a consistency issue then its ok

**Comment by ArchangelSDY** [comment]
@ArcturusZhang @leni-msft Could you take a look at this pr?

**Comment by leni-msft** [comment]
@ArchangelSDY The "API Readiness Check" failed. Is your service deployed to prod already?

**Comment by ArchangelSDY** [comment]
@leni-msft The manifest is still going through SDP rollout. Do we need to wait until it finishes before this can be merged?

**Comment by leni-msft** [comment]
@ArchangelSDY yes, swagger cannot be merged until the API is ready. Merging swagger to main means the API is released, so can be consumed by downstream users.

**Comment by juniwang** [comment]
Hi @leni-msft any more comments for this PR?

**Comment by juniwang** [comment]
the new api-version is ready on Azure, could u plz try again?

**Comment by ArchangelSDY** [comment]
@leni-msft  CI passed.

**Review by leniatgh — APPROVED** [review]

**Pull request merged**
